mono/eglib/gfile-win32.c

```diff
@@ -114,11 +114,8 @@ g_file_test (const gchar *filename, GFileTest test)
 	}
 
 	if ((test & G_FILE_TEST_IS_EXECUTABLE) != 0) {
-		size_t len = strlen (filename);
-		if (len > 4 && strcmp (filename + len-3, "exe"))
-		    return TRUE;
-		    
-		return FALSE;
+		/* Testing executable permission on Windows is hard, and this is unused, treat as EXISTS for now. */
```

Comment:
I'd strike `for now`. I don't see it ever changing. Granted msvcrt `stat` does `strcmp`. Does coreclr have anything similar?